src/librustc_mir/build/mod.rs

```diff
@@ -341,6 +341,11 @@ impl<'a,'tcx> Builder<'a,'tcx> {
             }
         }
     }
+
+    pub fn check_overflow(&self) -> bool {
```

Comment:
@eddyb 

I am quite sure that overflow checking should go *before* MIR lowering, not *after* - this is a departure from today.